improve the "installing kubeflow" page

thesuperzapper opened the pull request:
This PR improves the `Getting Started / Installing Kubeflow` page in the following ways:

- Splits the distributions tables into two sections:
   - "Active Distributions" - any distribution which has had an update within the last 6 months
   - "Legacy Distributions" - any not active distribution, currently only has:
       -  `Argoflow` (Kubeflow 1.4)
       - `Kubeflow on Azure` (Kubeflow 1.2)
- Improves the readability of the distribution tables:
    - ~~Removed the "version" column as it was quite messy, and often had incorrect information as distributions forgot to update it~~ (replaced with "latest release" column)
    - Combines the distribution "links" into a single column (preparation for once https://github.com/kubeflow/website/issues/3074 is resolved)
- Added a link to the "official website" of each distribution
- Changes the names of the following distributions to be more correct:
    - `Kubeflow on OpenShift` --> `Open Data Hub`
    - `Kubeflow on Oracle Container Engine for Kubernetes` --> `Kubeflow on Oracle Cloud Infrastructure`
- Improves the section header names:
    - `Install a packaged Kubeflow distribution` --> `Packaged Distributions of Kubeflow`
    - `Install the Kubeflow Manifests manually` --> `Raw Kubeflow Manifests`
- Improves the wording of warnings and other text.

## Screenshot of the updated page:

![Installing-Kubeflow-Kubeflow](https://user-images.githubusercontent.com/5735406/205757476-3056658b-2f5b-44fb-8a72-7c4cc8b28774.png)

jbottum commented:
@VaishnaviHire Is there someone from the OpenShift team that would like to comment on this proposal ?

annajung commented on content/en/docs/started/installing-kubeflow.md:
```diff
@@ -9,140 +9,190 @@ Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it prov
         <td>Charmed Kubeflow</td>
         <td>Canonical</td>
-        <td>Conformant Kubernetes</td>
-        <td>1.6</td>
```
I'm not sure if there was any previous discussion on this. Could you share the reason for removing the latest version supported by the distributions? I feel that this is the most important info on this page for the community to determine which distributions are "up to date." I think it's super valuable to have a page that shows lists of distributions + the latest version they support.

thesuperzapper replied:
@annajung previously I agreed with you, in fact, I made this page initially and was the one to add the version!

However, I now think separating by "active" vs "inactive" is more useful to users (especially new ones).

For most users, this is the first page they see when they want to start using Kubeflow, the goal of this page is simply to show them the options they have to use kubeflow in their environment.

----

When 1.7 is released, we can consider adding a second "active" table, which lists "distributions supporting Kubeflow 1.7 or later". 

However, this may not be necessary, as being "active" should imply they will support new versions in a timely manner.

DomFleischmann replied:
I agree with @annajung here, the version that each distribution supports seems to be one of the most important information the table provides. There also might be distributions who do active maintenance and support of an older version and might not update on time. Letting users know that they might get an older version with those distributions is important.

thesuperzapper replied:
@DomFleischmann Currently, this PR has not reduced the amount of version information, because all the "active" distributions support Kubeflow 1.6, and the heading above the table says `"distributions that provide Kubeflow 1.6 or later"`.

When 1.7 is released, we can consider adding a second "active" table, which lists `"distributions supporting Kubeflow 1.7 or later"`. However, this may not be necessary, as being "active" should imply they will support new versions in a timely manner.

annajung replied:
Maybe @kimwnasptd can chime in here, but Arrikto currently does not support 1.6+, according to https://www.kubeflow.org/docs/started/installing-kubeflow/. They're still on 1.4, so the changes here wouldn't make sense for Arrikto to be part of the "active" list.

thesuperzapper replied:
@andreeamun we need to confirm with Arrikto, what versions are supported by "Arrikto Kubeflow as a Service" and "Arrikto Enterprise Kubeflow", because I think the table was out of date.

@akgraner can you please confirm?

thesuperzapper replied:
As part of what I changed in https://github.com/kubeflow/website/pull/3381#issuecomment-1309845940, I have now removed any mention of specific Kubeflow versions, so it is less important to confirm the exact "versions" included in Arritko's distributions.

annajung commented on content/en/docs/started/installing-kubeflow.md:
```diff
@@ -9,140 +9,190 @@ Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it prov
-        <td></td>
+        <td>Amazon Web Services</td>
+        <td>
+          ➤ Amazon Elastic Kubernetes Service (EKS)
```
nit: ➤ symbol can be mistaken as a clickable drop-down

Only Arrikto has three listed, can we remove this and have bullet points for Arrikto only?

thesuperzapper replied:
@annajung I agree that the  `➤` symbol was confusing (and messy).

I have removed the "target platform" bullets for all but the Arrikto one and used a smaller bullet `◦` for that case.

I have updated the "links" bullets to use `⊛` for the "official website" and `◌` for the "other docs".

annajung commented on content/en/docs/started/installing-kubeflow.md:
```diff
@@ -9,140 +9,190 @@ Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it prov
+
+### Legacy Distributions
+
+The following table lists some <b>distributions which provide versions older than `Kuebflow 1.6`</b>.
```
```suggestion
The following table lists some <b>distributions which provide versions older than `Kubeflow 1.6`</b>.
```

thesuperzapper replied:
Thanks, I have resolved this typo in https://github.com/kubeflow/website/pull/3381/commits/79e0d92d3b5e9054a2ed84e84b8fdf3fda87b08c.

annajung reviewed (commented):
Thanks for working on this! I think it makes sense to move the distributions that are no longer supported into the "legacy" section. 

Left a few comments on a few concerns with what defines as "active" vs "legacy" and version removal

thesuperzapper commented on content/en/docs/started/installing-kubeflow.md:
```diff
@@ -9,140 +9,190 @@ Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it prov
-<b>See the table below for a list of options and links to documentation:</b>
+### Active Distributions
+
+The following table lists some <b>distributions that provide <a href="/docs/releases/kubeflow-1.6/"><code>Kubeflow 1.6</code></a> or later</b>.
```
@annajung the initial goal is just about separating the clearly inactive distributions (as they have had no updates for 12+ months)

In the future, we can agree on a metric, for example, if there have been no releases in 6+ months, or the project is clearly abandoned (has no maintainers).

thesuperzapper replied:
As stated in https://github.com/kubeflow/website/pull/3381#issuecomment-1309845940, I have now updated the definition of "active" to be "had a release in the last 6 months".

thesuperzapper commented:
Quick update, I have improved the bullet point symbols after the discussion in https://github.com/kubeflow/website/pull/3381#discussion_r999974283, see the screenshot at the top of the page.

andreeamun commented:
Hello:) It is great you addressed this. One thing that has been raised in the past is the order that distributions are listed (based on this proposal in the Active distributions one). Is there any reason behind this?

Might be more efficient to list them alphabetically.

thesuperzapper commented:
> Hello:) It is great you addressed this. One thing that has been raised in the past is the order that distributions are listed (based on this proposal in the Active distributions one). Is there any reason behind this?
> 
> Might be more efficient to list them alphabetically.

@andreeamun I have left the order as it was for this PR, I think we can discuss the best (and more fair) ordering in a separate discussion.

I also think we need to confirm the "names" of distributions. The current names stem from a time when the community owned all the distributions. Now that the distributions are maintained externally, having them all called `Kubeflow on XXXX` is problematic (and hurts distributions not named like this) because it implies that it is the "official" way to deploy Kubeflow on XXXX. 

Perhaps we should encourage distributions to name themselves something completely different, or suggest `Kubeflow by YYYYY`, where `YYYYY` is the maintainer of the distribution (for example, `Kubeflow by Google Cloud` or `Kubeflow by AWS`).

thesuperzapper commented:
@zijianjoy and @annajung what do you think we need to get this merged?

I think the overall changes make the "installing kubeflow" page much easier to understand.

annajung commented:
@thesuperzapper I am still concerned about the removal of the versions. Without the versions listed, tracking which Kubeflow versions each distribution support would be harder. In the future, how would we be able to determine which distributions are active vs. inactive without adding the manual work to check their external doc? Wouldn't it be better to keep this page as the source of truth and work with distributions to update it?

In addition, there is an [open question](https://github.com/kubeflow/website/pull/3381#discussion_r1004684299) about if all distributions listed in the "active" section support 1.6 that needs to be resolved before we can merge this PR. 

If the [initial goal](https://github.com/kubeflow/website/pull/3381#discussion_r1000012520) is to separate active and inactive, I think you can still move forward by keeping the versions as is and just separating the distributions that support 1.3 and below (more than 12months+). wdyt?

jbottum commented:
Some comments, 1) I expect we will have additional distributions: VMware, HP, Civo, DKube/OneConvergence,  2) as the list gets longer, it will be harder to keep updated, especially if we are displaying version numbers.    3) I like version numbers, but in reality, distributions will modify (so does the version number mean as much i.e. Arrikto distribution has things that might eclipse 1.6, even if they publicly stating on 1.x).

thesuperzapper commented:
@annajung @jbottum I have updated the PR with what I think is a good compromise.

1. I have updated the definition of "active" to be "having a release within the last 6 months"
2. I have added a "latest release" column to the table, which lists the date (with a citation) of the most recent release

In the future when we have "conformance testing" (https://github.com/kubeflow/kubeflow/issues/6485), we can list which version of the Kubeflow "API" they are conformant with, which is more useful than listing "Kubeflow 1.X" as that is meaningless when we allow vendors to change anything they like about Kubeflow.

thesuperzapper commented:
> @thesuperzapper @annajung Side note - perhaps instead of list distributions in alphabetical order, we list in order of date of 1st supported. This rewards the distributions that have been in the Community the longest and put them at the top of the list. Of course this assumes they have kept up with the subsequent releases.

@jbottum I think let's leave the re-ordering discussion to a separate PR, so we can get this update merged soon.

Note, I have left the order unchanged in this PR (other than moving some distributions to the new "legacy" table).

thesuperzapper commented:
I just did a quick formatting improvement in https://github.com/kubeflow/website/pull/3381/commits/b1287ad8f5fe1d2e9dcfb6efc4ec66ddb6bf35f9:

1. Used fewer words (to reduce the risk of text wrapping on smaller screens)
     - "Official Website" -> "Website"
     - "References" -> "Links"
2. Added line breaks after the date for the references in "Latest Release" (reduces column width)
3. Removed some unnecessary capital letters

annajung commented on content/en/docs/started/installing-kubeflow.md:
```diff
@@ -9,140 +9,249 @@ Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it prov
+          <sub><u><a href="/docs/distributions/ibm/">Legacy Docs</a></u></sub>
+        </td>
+        <td>
+          Unknown
```
@yhwang is there something we can reference here to add the date of your latest release?

yhwang replied:
thanks for bringing this to my attention. the latest release date is `2022-10-10` for IKS

thesuperzapper replied:
> thanks for bringing this to my attention. the latest release date is `2022-10-10` for IKS

@yhwang can you share a public announcement, GitHub release, or changelog that has the release date on it? We need a reference so we can link it.

yhwang replied:
https://github.com/IBM/manifests/releases/tag/v1.6.1

thesuperzapper replied:
@yhwang I have updated the table to reflect this information in https://github.com/kubeflow/website/pull/3381/commits/98d74badfafe3c35ddb56ec3d408129d29ff3cdf

yhwang replied:
thanks

annajung commented on content/en/docs/started/installing-kubeflow.md:
```diff
@@ -9,140 +9,249 @@ Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it prov
+          <u><a href="https://www.arrikto.com/kubeflow-as-a-service/">Website</a></u>
+        </td>
+        <td>
+          Unknown
```
@kimwnasptd is there something we can reference here to add the date of your latest release?

thesuperzapper replied:
@kimwnasptd just to clarify, we need a publicly listed blog/changelog/release-page that confirms the release date.

For example, I have used https://docs.arrikto.com/Changelog.html for the "Arrikto Enterprise Kubeflow" distribution, but "Arrikto Kubeflow as a Service" dose not seem to have an equivalent page.

annajung commented on content/en/docs/started/installing-kubeflow.md:
```diff
@@ -9,140 +9,249 @@ Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it prov
+          <u><a href="https://github.com/oracle-devrel/kubeflow-oke">Website</a></u>
+        </td>
+        <td>
+          Unknown
```
@julioo is there something we can refer here to add the date of your latest release?

thesuperzapper replied:
@julioo just to clarify, we need a publicly listed blog/changelog/release-page that confirms the release date so we can link it.

julioo replied:
@annajung @thesuperzapper, you're referring to the Oracle Kubernetes Engine release date?
https://docs.oracle.com/en-us/iaas/releasenotes/services/conteng/
for example
https://docs.oracle.com/en-us/iaas/releasenotes/changes/2aca54a0-5594-48f3-8c51-c126d22d4eda/

annajung reviewed (commented):
If the kubeflow versions are replaced with the distributions' latest release date, can we make sure that all distributions list the dates?

tagging distribution owners for unknown values

thesuperzapper commented:
I wanted to share another proposal for how to order the "installing kubeflow" table.
_(NOTE: I still think we should leave the order as is in this PR and do any re-order in a separate one)_

I think we should sort the table alphabetically, but have separate tables for each "tier of partner" (think "platinum", "gold", and "silver"). Partners would attain tiers by accumulating points through their contributions and involvement in the project.

For example, some things which could give you points might be:

- length of involvement in the project
- number of FTE committed to the project (and how active they are)
- other contributions (like providing food at events, sponsoring testing infrastructure, etc)
- how consistently you upstream proprietary features back to the community
- the openness of your distribution (for example, less points for closed-off distributions, and more for fully open-source)

_We will probably need to treat "non-commercial" (free, open source) partners with a different set of standards so that we don't end up with commercial distributions crowding them out._

----

Separately, we will need to rename distributions which are called things like `Kubeflow on XXXXX` as this is very misleading for new users who will believe that this is the only way to deploy on XXXXX (as XXXXX is often a platform, like AWS). 

In most cases, a rename to something like `XXXXX Distribution of Kubeflow` will be acceptable (like `Google Cloud Distribution of Kubeflow` already has), but I would also love to see some more exotic names which don't have the word Kubeflow in it (like `Open Data Hub`)!

jbottum commented:
@johnugeorge Hey Johnu, this update show that Nutanix does not have a current distribution of Kubeflow.    Would you please review and provide the correct info ?

jbottum commented:
@thesuperzapper Hi Mathew - I believe your proposal on ordering and levels (especially using sweat equity) has merit.    Per your/my earlier comments, I think that the re-ordering (and the distribution requirement restructuring) need to be in separate PRs.      In the short term (by the end of the month or sooner), I propose that we need to add Oracle, Telekom, and Vmware and make sure the info listed for each distribution is correct.    Question - Should we require a looks good to me from each of the distribution contacts before this is merged ?

johnugeorge commented:
@thesuperzapper Nutanix supports Kubeflow 1.6 and was announced already https://github.com/kubeflow/manifests/issues/2221#issuecomment-1230681202. Please move it to active distributions.

Ref: https://github.com/nutanix/karbon-platform-services/issues/94

On a larger note, what are we aiming to achieve with this change? This change needs more discussion before merge.

thesuperzapper commented:
> @thesuperzapper Nutanix supports Kubeflow 1.6 and was announced already [kubeflow/manifests#2221 (comment)](https://github.com/kubeflow/manifests/issues/2221#issuecomment-1230681202). Please move it to active distributions.
> 
> Ref: [nutanix/karbon-platform-services#94](https://github.com/nutanix/karbon-platform-services/issues/94)

@johnugeorge I have updated the table to reflect this information in https://github.com/kubeflow/website/pull/3381/commits/98d74badfafe3c35ddb56ec3d408129d29ff3cdf

_PS: I think it's a little strange to use a GitHub comment to announce a release, that's obviously up to Nutanix to decide how they do things._

> On a larger note, what are we aiming to achieve with this change? This change needs more discussion before merge.

This update is about a number of things (see the PR description), but primarily it's about making the tables more readable, and preparing for when this page will be the only reference to distributions on the `kubeflow.org` site (e.g. we will be linking to each distributions site, but not hosting their docs for them, see https://github.com/kubeflow/website/issues/3074).

thesuperzapper commented:
@zijianjoy all Netelify builds are now failing because (we need to update the build image in the Netelify admin page), do you have access to update it?

I have raised an issue to discuss this problem: https://github.com/kubeflow/website/issues/3404

gkcalat commented:
> 

@thesuperzapper, please push a new commit to trigger the tests

thesuperzapper commented:
@gkcalat thanks for that!

---

@annajung @liuqi  I have rebased for https://github.com/kubeflow/website/pull/3396, however, I can't find a public announcement/changelog for `Kubeflow on vSphere`, so I have put "unknown" as the "latest release date" for now.

Do you have a public place where you announced the release of `Kubeflow on vSphere` for Kubeflow 1.6? 

For example, others are using the "releases" page on GitHub, or their official blog posts (a big goal of this PR is to have each distribution provide a "latest release date" with a link as evidence for this date).

---

For @zijianjoy and everyone watching, here is a screenshot from this PR (as it currently stands).

![Installing-Kubeflow-Kubeflow](https://user-images.githubusercontent.com/5735406/205758300-cd9cc5ad-4df2-44d4-b9bf-a07bfb93084e.png)
